Framework/Core/include/Framework/ASoA.h

```diff
@@ -763,35 +786,35 @@ class Table
     using RowViewCore<IP, C...>::operator++;
 
     /// Allow incrementing by more than one the iterator
-    RowViewBase<IP, T...> operator+(int64_t inc) const
+    RowViewBase operator+(int64_t inc) const
     {
-      RowViewBase<IP, T...> copy = *this;
+      RowViewBase copy = *this;
       copy.moveByIndex(inc);
       return copy;
     }
 
-    RowViewBase<IP, T...> operator-(int64_t dec) const
+    RowViewBase operator-(int64_t dec) const
     {
       return operator+(-dec);
     }
 
-    RowViewBase<IP, T...> const& operator*() const
+    RowViewBase const& operator*() const
     {
       return *this;
     }
   };
-  template <typename... Ts>
-  using RowView = RowViewBase<DefaultIndexPolicy, Ts...>;
+  template <typename P, typename... Ts>
+  using RowView = RowViewBase<DefaultIndexPolicy, P, Ts...>;
 
-  template <typename... Ts>
-  using RowViewFiltered = RowViewBase<FilteredIndexPolicy, Ts...>;
+  template <typename P, typename... Ts>
+  using RowViewFiltered = RowViewBase<FilteredIndexPolicy, P, Ts...>;
 
-  using iterator = RowView<table_t>;
-  using const_iterator = RowView<table_t>;
-  using unfiltered_iterator = RowView<table_t>;
-  using unfiltered_const_iterator = RowView<table_t>;
-  using filtered_iterator = RowViewFiltered<table_t>;
-  using filtered_const_iterator = RowViewFiltered<table_t>;
+  using iterator = RowView<table_t, table_t>;
+  using const_iterator = RowView<table_t, table_t>;
+  using unfiltered_iterator = RowView<table_t, table_t>;
+  using unfiltered_const_iterator = RowView<table_t, table_t>;
+  using filtered_iterator = RowViewFiltered<table_t, table_t>;
+  using filtered_const_iterator = RowViewFiltered<table_t, table_t>;
```

Comment:
Why twice?

Reply:
You mean twice the the template parameter? One is the parent type, the other is the original. Directly declared soa::Table is the original of itself, for the derived things like Joins it is redefined as `RowView<Join<T...>, T...>`